package.json

```diff
@@ -24,18 +24,22 @@
   },
   "type": "module",
   "main": "dist/docsify.js",
+  "types": "src/core/Docsify.d.ts",
   "exports": {
     ".": "./src/core/Docsify.js",
     "./*": "./*"
   },
   "files": [
-    "dist"
+    "dist",
+    "src"
```

Comment:
The src folder has been removed.

Will adding it back now affect users?

Reply:
This is how we can give people type definitions with good DX (f.e. `Go To Definition` works well) with the original ESM files. This also gives people with more flexibility in configuring how they import and/or build Docsify into their app (build example not shown in this PR).

Reply:
Try opening the `test/consume-types` folder in VS Code as the root folder (f.e. `code test/consume-types`), then run `npm clean-install --install-links` (the `--install-links` option will take the Docsify source from the root of the repo, build it, and install it as if it were downloaded from the npm registry). Finally you can open up the `example.js` file and try out type checking and intellisense (f.e. try looking at config option types, or running `Go To Definition` on `Docsify`).

Reply:
Here's a video showing the intellisense:


https://github.com/user-attachments/assets/b0214aa5-abb0-422a-a8bd-56bb2dcc791d